ChatRPG/Services/GameInputHandler.cs

```diff
@@ -60,81 +66,113 @@ private void OnChatCompletionChunkReceived(bool isStreamingDone, string? chunk =
+                _logger.LogInformation("Hurt or heal response: {hurtOrHealString}", hurtOrHealString);
+                OpenAiGptMessage hurtOrHealMessage = new(ChatMessageRole.Assistant, hurtOrHealString);
+                LlmResponse? hurtOrHealResponse = hurtOrHealMessage.TryParseFromJson();
+                string hurtOrHealMessageContent = "";
+                Random rand = new Random();
+                if (hurtOrHealResponse?.Heal == true)
                 {
-                    messageContent +=
-                        $" With no health points remaining, {opponent.Name} dies and can no longer participate in the narrative.";
+                    int healAmount = rand.Next(PlayerHealMin, PlayerHealMax);
+                    campaign.Player.AdjustHealth(healAmount);
+                    hurtOrHealMessageContent +=
+                        $"The player heals {healAmount} health points, setting them at {campaign.Player.CurrentHealth} health points. Mention these numbers in your response.";
                 }
-                messageContent += $"The player hits with their attack, dealing {playerDmg} damage. The opponent has {opponent.CurrentHealth} health remaining.";
-                _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", campaign.Player.Name, opponent.Name, playerDmg, opponent.CurrentHealth, opponent.MaxHealth);
-            }
-            else
-            {
-                messageContent += $"The player misses with their attack, dealing no damage. The opponent has {opponent.CurrentHealth} health remaining.";
-            }
 
-            if (opponentDmg != 0)
-            {
-                messageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has {campaign.Player.CurrentHealth} health remaining.";
-                if (campaign.Player.AdjustHealth(-opponentDmg))
+                if (hurtOrHealResponse?.Hurt == true)
                 {
-                    await HandlePlayerDeath(campaign.Player, conversation);
+                    int dmgAmount = rand.Next(PlayerDmgMin, PlayerDmgMax);
+                    hurtOrHealMessageContent += $"The player hurts themselves for {dmgAmount} damage. The player has {(campaign.Player.CurrentHealth - dmgAmount < 0 ? 0 : campaign.Player.CurrentHealth - dmgAmount)} health remaining. Mention these numbers in your response.";
+                    if (campaign.Player.AdjustHealth(-dmgAmount))
+                    {
+                        hurtOrHealMessageContent += "The player has died and their adventure ends.";
+                    }
+                }
+                OpenAiGptMessage hurtOrHealSystemMessage = new(ChatMessageRole.System, hurtOrHealMessageContent);
+                conversation.Add(hurtOrHealSystemMessage);
+                return _systemPrompts[SystemPromptType.DoAction];
+            case UserPromptType.Attack:
```

Comment:
Lav en private method til at håndtere Do. Måske også små methods til at sørge for heals og hurts.

Reply:
ye